apps/omg_watcher_info/lib/omg_watcher_info/db/eth_event.ex

```diff
@@ -234,6 +237,69 @@ defmodule OMG.WatcherInfo.DB.EthEvent do
     {:ok, get(ethevent.root_chain_txhash_event)}
   end
 
+  # preload txoutputs in a single query as there will not be a large number of them
+  def get(root_chain_txhash_event) do
+    DB.Repo.one(
+      from(ethevent in base_query(),
+        where: ethevent.root_chain_txhash_event == ^root_chain_txhash_event
+      )
+    )
+  end
+
+  @doc """
+  Gets a paginated list of events, optionally filtered by address and/or type.
+  """
+  @spec get_events(
+          paginator :: Paginator.t(%DB.EthEvent{}),
+          event_type :: atom() | nil,
+          address :: Crypto.address_t() | nil
+        ) :: Paginator.t(%DB.EthEvent{})
+  def get_events(paginator, event_type \\ nil, address \\ nil) do
```

Comment:
what events are being fetched at the moment apart from :deposits? why are there optional arguments available, if we're not using them in production code?
I see that we're always passing in event_type (:deposit) and address (even though it can be nil).

So the question is for whom was this function written?
What are the `event_type` types, and why aren't we able to define them in dialyzer spec?

Reply:
An `address` parameter is optional for the same reason that it is for `transaction.all`: ability to get a paginated list of _all_ deposits if no parameter is passed. 

A query for `event_type` would be re-usable for a future `/exit.all` endpoint. This does not exist yet but the intention was to write with composability and re-usability in mind for this and other future features. 

That being said, I've updated the dialyzer spec with a defined set for `event_types` and also refactored to remove defaulting to `nil` in `get_events` since we want to avoid default arguments.

Reply:
Is there a requirement to get everyone's deposits without specifying an address? if not, remove it. The API needs to be strict - fulfil the product requirement and nothing more. These queries are expensive and may harm us more then give us. If there's ever a product requirement to fetch all deposits (is there already?) we can revisit. Otherwise, we write code per requirements.

Similar to the above. If there's immediate requirement - ticket with description that this is needed otherwise we do not write future code.

Are event_types sanitised anywhere?

Reply:
Perhaps this is new to you, so I’ll briefly explain why future proof code is bad:
- we have more stuff to maintain while we wait for a use case for the code that was written, perhaps that use case never arrives or perhaps we shifted focus, perhaps none of our partners want that feature

- code without actual use case is not tested by our acceptance tests, because it’s not based from features we need

- refactoring is harder, because reading code you’re looking at things through lense of usage, if there is none... it’s difficult to reason

- theres a chance future use case code is less safe because it has less restrictions

Reply:
Hey Ino. Thanks for your perspective. I especially agree that returning _all_ deposits with no address filter is not a conceivable use case, so I've made the address parameter non-optional from the controller downwards (and updated the Swagger spec)

On the above snippet, I've also scoped `query_by_type` to `query_deposits`. The code stays is composable and re-usable "enough" for the an `exit.all` or `standard_exit.all` endpoint. (This is an existing ticket which I hope to take later this cycle) 

This removes the need for `event_type` sanitisation.